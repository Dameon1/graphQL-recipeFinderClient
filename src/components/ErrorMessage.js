import React from "react";
import { ErrorStyles } from "./styles";
import PropTypes from "prop-types";

const ErrorMessage = ({ error }) => {
  if (!error || !error.message) return null;
  if (
    error.networkError &&
    error.networkError.result &&
    error.networkError.result.errors.length
  ) {
    return error.networkError.result.errors.map((error, i) => (
      <ErrorStyles key={i}>
        <p data-test="graphql-error">
          <strong>Shoot!</strong>
          {error}
        </p>
      </ErrorStyles>
    ));
  }
  return (
    <ErrorStyles>
      <p data-test="graphql-error">
        <strong>Shoot!</strong>
        {error.message}
      </p>
    </ErrorStyles>
  );
};

ErrorMessage.defaultProps = {
  error: {}
};

ErrorMessage.propTypes = {
  error: PropTypes.object
};

export default ErrorMessage;
