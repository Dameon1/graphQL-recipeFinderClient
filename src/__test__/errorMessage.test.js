import React from "react";
import ErrorMessage from "../components/ErrorMessage";
import { mount } from "enzyme";
import toJSON from "enzyme-to-json";

const errorMessage = "simple error message";

describe("<ErrorMessage/>", () => {
  it("displays the error message given", async () => {
    const wrapper = mount(<ErrorMessage error={{ error: errorMessage }} />);
    wrapper.update();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
