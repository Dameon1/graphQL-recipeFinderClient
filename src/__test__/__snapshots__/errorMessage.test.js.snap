// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage/> displays the error message given 1`] = `
<ErrorMessage
  error={
    Object {
      "error": "simple error message",
    }
  }
/>
`;
